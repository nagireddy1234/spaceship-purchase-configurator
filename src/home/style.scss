@import "../theme/colors.scss";
.wrapper {
  background-color: $blue;
  width: 100%;
  min-height: 100vh;
  height: 100%;
  padding: 2rem 0;
  .card-wrapper {
    border: 1px solid $lighGreen;
    border-radius: 0.5rem;
    max-width: 59rem;
    margin: auto;
    padding: 2.5rem 2rem 2.5rem 2rem;
    width: 100%;
    .heading {
      font-family: Roboto Mono;
      color: $lighGreen;
      font-style: normal;
      font-weight: normal;
      font-size: 1.5rem;
      line-height: 2rem;
      text-align: center;
      letter-spacing: 0.2em;
      margin-bottom: 2.812rem;
    }
    .row {
      display: flex;
      align-items: flex-start;
      justify-content: space-between;
    }
    .w-80 {
      width: 80%;
    }
    .tool-name {
      color: $lighGreen;
      font-size: 1.125rem;
      margin-bottom: 0.5rem;
    }
    div.selected {
      border: 1px solid $lighGreen;
      color: $lighGreen;
      filter: drop-shadow(0px 0px 10px #7bf762);
    }
    .col-4,
    .col-3 {
      width: calc(33.33% - 3.5rem);
      border: 1px solid $fadeGreen;
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      border-radius: 0.5rem;
      padding: 1rem;
      cursor: pointer;
      color: $fadeGreen;
    }
    .col-3 {
      width: calc(33.33% - 3.5rem);
    }
    .col-80 {
      width: calc(100% - 2rem);
      display: flex;
      justify-content: space-between;
    }
    .spacing {
      width: 100%;
      height: 3rem;
    }
    .w-20 {
      width: calc(25% - 1rem);
      min-width: 19rem;
      margin-top: 1.95rem;
      .info-card {
        background-color: $darkGreen;
        padding: 1rem;
        border-radius: 0.5rem;
        border: 1px solid $lighGreen;
        .flex {
          display: flex;
          align-items: center;
          justify-content: space-between;
          padding: 0.5rem;
        }
        .info-title {
          font-size: 0.875rem;
          color: $lighGreen;
          font-style: normal;
          font-weight: normal;
          letter-spacing: 0.2em;
        }
        .title-price {
          color: $white;
          font-size: 0.875rem;
          font-style: normal;
          font-weight: normal;
          letter-spacing: 0.2em;
        }
        .divider {
          height: 1px;
          width: 100%;
          background-color: $lighGreen;
          margin-top: 0.937rem;
          margin-bottom: 1.437rem;
        }
        .title {
          font-size: 0.875rem;
        }
        .total-price,
        .info-total {
          font-size: 1.125rem;
          font-family: Roboto Mono;
          font-style: normal;
          font-weight: normal;
          color: $lighGreen;
          letter-spacing: 0.2em;
        }
        .total-price {
          color: $white;
        }
      }
    }
  }
}

@media screen and (min-width: 576px) and (max-width: 1062px) {
  .wrapper {
    padding: 1rem;
    .card-wrapper {
      padding: 2rem;
      width: auto;
      max-width: fit-content;
      margin: 1rem auto;
      .row {
        flex-direction: column;
      }
      .w-80 {
        width: 100%;
      }
      .col-80 {
        width: 100%;
      }
      .w-20 {
        width: 100%;
        margin-top: 4.5rem;
      }
      .col-4 {
        width: calc(33.33% - 3.5rem);
      }
      .col-3 {
        width: calc(33.33% - 1.5rem);
      }
    }
  }
}
@media screen and (min-width: 0px) and (max-width: 575px) {
  .wrapper {
    .card-wrapper {
      padding: 0.5rem;
      width: auto;
      border: 0;
      max-width: fit-content;
      margin: 0.5rem auto;
      .row {
        flex-direction: column;
      }
      .w-80 {
        width: 100%;
      }
      .col-80 {
        width: 100%;
      }
      .w-20 {
        width: 100%;
        margin-top: 4rem;
      }
      .col-4 {
        width: calc(33.33% - 2.5rem);
      }
      .col-3 {
        width: calc(33.33% - 0.5rem);
      }
    }
  }
}
