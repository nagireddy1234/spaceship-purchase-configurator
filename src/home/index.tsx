import { useState } from "react";
import ColorSelector from "../components/colorSelector";
import OptionPackage from "../components/optionPackage";
import ToolCard from "../components/toolCard";
import { basicPackage, luxPackage, sportPackage } from "../data/packeges";
import { colorPrice, optionPrice, powerPrice, warpPrice } from "../data/price";
import "./style.scss";

const colors = {
  white: "#fff",
  orange: "#FF7A00",
  skyeblue: "#6BE4FF",
};

const Home = () => {
  const [state, setState] = useState({
    colorSelected: -1,
    powerSelected: -1,
    warpSelected: -1,
    optionalSelected: -1,
  });

  const [prices, setPrices] = useState({
    basePrice: 1000,
    colorPrice: 0,
    powerPrice: 0,
    warpPrice: 0,
    optionPrice: 0,
  });

  const handleSelect = (type: number, keyName: string) => {
    const newState: any = { ...state };
    newState[keyName] = type;
    setState(newState);
    const newPrices = { ...prices };
    if (keyName === "colorSelected") {
      newPrices.colorPrice = colorPrice[type - 1];
    }
    if (keyName === "powerSelected") {
      newPrices.powerPrice = powerPrice[type - 1];
    }
    if (keyName === "warpSelected") {
      newPrices.warpPrice = warpPrice[type - 1];
    }
    if (keyName === "optionalSelected") {
      newPrices.optionPrice = optionPrice[type - 1];
    }
    setPrices(newPrices);
  };

  return (
    <div className="wrapper">
      <div className="card-wrapper">
        <h4 className="heading">Spaceship configurator</h4>
        <div className="row">
          <div className="w-80">
            <p className="tool-name">Select color:</p>
            <div className="col-80">
              <ColorSelector
                onClick={() => handleSelect(1, "colorSelected")}
                selected={state.colorSelected === 1}
                color={colors.white}
                colorName="White"
                price={0}
              />
              <ColorSelector
                selected={state.colorSelected === 2}
                onClick={() => handleSelect(2, "colorSelected")}
                color={colors.orange}
                colorName="Volcano"
                price={100}
              />
              <ColorSelector
                selected={state.colorSelected === 3}
                onClick={() => handleSelect(3, "colorSelected")}
                color={colors.skyeblue}
                colorName="Sky"
                price={100}
              />
            </div>
            <div className="spacing" />
            <p className="tool-name">Select power:</p>
            <div className="col-80">
              <ToolCard
                title="100 MW"
                price={0}
                selected={state.powerSelected === 1}
                onClick={() => handleSelect(1, "powerSelected")}
              />
              <ToolCard
                title="150 MW"
                price={200}
                selected={state.powerSelected === 2}
                onClick={() => handleSelect(2, "powerSelected")}
              />
              <ToolCard
                title="200 MW"
                price={500}
                selected={state.powerSelected === 3}
                onClick={() => handleSelect(3, "powerSelected")}
              />
            </div>
            <div className="spacing" />
            <p className="tool-name">Warp drive:</p>
            <div className="col-80 justify-start">
              <ToolCard
                title="NO"
                price={0}
                selected={state.warpSelected === 1}
                onClick={() => handleSelect(1, "warpSelected")}
              />
              <ToolCard
                title="YES"
                price={1000}
                selected={state.warpSelected === 2}
                onClick={() => handleSelect(2, "warpSelected")}
              />
              <div className="col-4" style={{ visibility: "hidden" }} />
            </div>
            <div className="spacing" />
            <p className="tool-name">Select option package:</p>
            <div className="col-80 justify-start">
              <OptionPackage
                planName="Basic"
                hidePrice
                data={basicPackage}
                selected={state.optionalSelected === 1}
                onClick={() => handleSelect(1, "optionalSelected")}
                price={0}
              />
              <OptionPackage
                planName="Sport"
                price={100}
                data={sportPackage}
                selected={state.optionalSelected === 2}
                onClick={() => handleSelect(2, "optionalSelected")}
              />
              <OptionPackage
                planName="Lux"
                price={500}
                data={luxPackage}
                selected={state.optionalSelected === 3}
                onClick={() => handleSelect(3, "optionalSelected")}
              />
            </div>
          </div>
          <div className="w-20">
            <div className="info-card">
              <div className="flex">
                <p className="info-title">Base price:</p>
                <p className="title-price">+{prices.basePrice}€</p>
              </div>
              <div className="flex">
                <p className="info-title">Color:</p>
                <p className="title-price">+{prices.colorPrice}€</p>
              </div>
              <div className="flex">
                <p className="info-title">Power:</p>
                <p className="title-price">+{prices.powerPrice}€</p>
              </div>
              <div className="flex">
                <p className="info-title">Warp drive:</p>
                <p className="title-price">+{prices.warpPrice}€</p>
              </div>
              <div className="flex">
                <p className="info-title">Option package:</p>
                <p className="title-price">+{prices.optionPrice}€</p>
              </div>
              <div className="divider"></div>
              <div className="flex">
                <p className="info-total">Total:</p>
                <p className="total-price">
                  {prices.basePrice +
                    prices.colorPrice +
                    prices.optionPrice +
                    prices.powerPrice +
                    prices.warpPrice}
                  €
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
