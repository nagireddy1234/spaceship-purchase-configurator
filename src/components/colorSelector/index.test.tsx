import React from 'react';
import renderer from 'react-test-renderer';
import { colors } from '../../home';
import ColorSelector from './index';

it('Checking ColorSelector component rendring correctly or not.', () => {
    const home = renderer
        .create(<ColorSelector onClick={() => {}} selected={false} color={colors.white} colorName="White" price={0} />)
        .toJSON();
    expect(home).toMatchSnapshot();
});
