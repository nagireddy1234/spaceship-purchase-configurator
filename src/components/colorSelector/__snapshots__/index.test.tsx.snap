// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking ColorSelector component rendring correctly or not. 1`] = `
<div
  className="col-4 "
  onClick={[Function]}
>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  />
  <p
    className="price"
  >
     +
    0
    â‚¬
  </p>
  <p
    className="colorname"
  >
    White
  </p>
</div>
`;
